flake.nix

```diff
@@ -0,0 +1,225 @@
+              });
+            })
+          ];
+        in {
+          inherit pythonOverrides;
+          # solc derivation is broken for darwin M1 in the version of nixpkgs we are on
+          # the derivation below is a copy from nixpkgs 22.05
+          solc = prev.callPackage ./package/nix/solc.nix { };
+
+          kevm = prev.stdenv.mkDerivation {
+            pname = "kevm";
+            version = self.rev or "dirty";
+            buildInputs = buildInputs final;
+            nativeBuildInputs = [ prev.makeWrapper ];
+
+            src = prev.stdenv.mkDerivation {
+              name = "kevm-${self.rev or "dirty"}-src";
+              src = prev.lib.cleanSource
+                (prev.nix-gitignore.gitignoreSourcePure [
+                  ./.gitignore
+                  ".github/"
+                  "result*"
+                  "*.nix"
+                  "deps/"
+                  "kevm_pyk/"
+                ] ./.);
+              dontBuild = true;
+
+              installPhase = ''
+                mkdir $out
+                cp -rv $src/* $out
+                chmod -R u+w $out
+                mkdir -p $out/deps/plugin
+                cp -rv ${prev.blockchain-k-plugin-src}/* $out/deps/plugin/
+              '';
+            };
+
+            dontUseCmakeConfigure = true;
+
+            patches = [ ./package/nix/kevm.patch ];
+
+            postPatch = ''
+              substituteInPlace ./cmake/node/CMakeLists.txt \
+                --replace 'set(K_LIB ''${K_BIN}/../lib)' 'set(K_LIB ${prev.k}/lib)'
+              substituteInPlace ./kevm \
+                --replace 'execute python3 -m kevm_pyk' 'execute ${final.python38Packages.kevm_pyk}/bin/kevm-pyk'
+            '';
+
+            buildFlags =
+              prev.lib.optional (prev.stdenv.isAarch64 && prev.stdenv.isDarwin)
+              "APPLE_SILICON=true";
+            enableParallelBuilding = true;
+
+            preBuild = ''
+              make plugin-deps
+            '';
+
+            installPhase = ''
+              mkdir -p $out
+              mv .build/usr/* $out/
+              wrapProgram $out/bin/kevm --prefix PATH : ${
+                prev.lib.makeBinPath [ final.solc ]
+              }
+              ln -s ${prev.k} $out/lib/kevm/kframework
+            '';
+          };
+
+          kevm-test = prev.stdenv.mkDerivation {
+            pname = "kevm-test";
+            version = self.rev or "dirty";
+
+            src = final.kevm.src;
+
+            enableParallelBuilding = true;
+
+            buildInputs = [ final.kevm prev.which prev.git ];
+
+            buildPhase = ''
+              mkdir -p tests/ethereum-tests/LegacyTests
+              cp -rv ${ethereum-tests}/* tests/ethereum-tests/
+              cp -rv ${ethereum-legacytests}/* tests/ethereum-tests/LegacyTests/
+              chmod -R u+w tests
+              APPLE_SILICON=${if prev.stdenv.isAarch64 && prev.stdenv.isDarwin then "true" else "false"} NIX=true package/test-package.sh
```

Comment:
This is pretty weird, what's going on here? Clone of this submodule happens out of tree and needs to  be copied in? Note that this submodule  is only used for testing, not building.